atc/creds/vault/vault.go

```diff
@@ -43,6 +51,15 @@ func (v Vault) NewSecretLookupPaths(teamName string, pipelineName string, allowR
 
 // Get retrieves the value and expiration of an individual secret
 func (v Vault) Get(secretPath string) (interface{}, *time.Time, bool, error) {
+	if v.LoggedIn != nil {
+		select {
+		case <-v.LoggedIn:
+			v.LoggedIn = nil
```

Comment:
Perhaps we should move this `select` anyway - if we fail to login to vault, there's no point blocking for 60 seconds every time we call `Get`

Reply:
@aoldershaw  Good point. If Vault login fails, `ReAuther.authLoop()` will keep retrying login to Vault. 

If the error is something like wrong password, then login will never succeed. In this case, I think there is no much difference between `Get` returning immediately with a "missing client token" error and returning an timeout error after blocking 60 seconds.

If Vault login failure is due to some temp problem, like flaky network, or Vault restart, or etc., then blocking might reduce failures. 

WDYT?

Reply:
I don't have strong opinion of setting `v.LoggedIn = nil`. I can delete it.

Reply:
True, good point - I'm cool with keeping the `select` here then, but if you could delete `v.LoggedIn = nil`, that'd be great.

---

I tested out passing invalid credentials to Vault both on `master` and on this PR. On `master`, we get a vault specific error:

```
failed to interpolate task config: Error making API request.

URL: GET https://vault:8200/v1/concourse/main/bar/foo
Code: 400. Errors:

* missing client token
```

and on this branch, after 60 seconds, we get:

```
failed to interpolate task config: timed out to login to vault
```

At first I was thinking the less descriptive error on this PR is an issue, but I realize now that it's actually *better*, since the `missing client token` error is misleading - when we can't login, we just use an unauthenticated client

Reply:
Deleted `v.LoggedIn = nil`.